src/ast/passes/codegen_llvm.cpp

```diff
@@ -2512,6 +2512,9 @@ void CodegenLLVM::visit(Predicate &pred)
 
   b_.CreateCondBr(expr_, pred_false_block, pred_true_block);
   b_.SetInsertPoint(pred_false_block);
+  if (need_recursion_check_) {
+    b_.CreateUnSetRecursion(current_attach_point_->loc);
+  }
```

Comment:
Let's push this down into createRet(). I have another PR that will make a call to createRet(). If someone later adds a createRet() but forgets to unset the recursion guard, it'll be disastrous.

Reply:
Ha.  I didn't even know `createRet` existed. I'm definitely putting up another PR to better distinguish this internal function from `CreateRet` from LLVM.